File: 算法/代码随想录算法PDF/00调试源代码/452.用最少数量的箭引爆气球.py
#
# @lc app=leetcode.cn id=452 lang=python3
#
# [452] 用最少数量的箭引爆气球
#

# @lc code=start
class Solution:
    def findMinArrowShots(self, points: List[List[int]]) -> int:
        if len(points) == 0: return 0
        points.sort(key = lambda x : x[0])
        res = 1
        for i in range(1, len(points)):
            if points[i][0] > points[i - 1][1]:
                res += 1
            else:
                points[i][1] = min(points[i][1], points[i-1][1])
        return res
# @lc code=end

File: 算法/代码随想录算法PDF/00调试源代码/452.用最少数量的箭引爆气球.cpp
/*
 * @lc app=leetcode.cn id=452 lang=cpp
 *
 * [452] 用最少数量的箭引爆气球
 */

// @lc code=start
class Solution {
public:
    static bool cmp(const vector<int>& a, const vector<int>& b) {
        return a[0] < b[0];
    }
    int findMinArrowShots(vector<vector<int>>& points) {
        if (points.size() == 0) return 0;
        sort(points.begin(), points.end(), cmp);
        int res = 1;  // 至少需要一个
        for (int i = 1; i < points.size(); i++) {
            if (points[i][0] > points[i - 1][1]) {
                res++;
            } else {
                points[i][1] = min(points[i - 1][1], points[i][1]);
            }
        }
        return res;
    }
};
// @lc code=end

